bentoml/_internal/bento/gen.py

```diff
@@ -115,71 +122,85 @@ def get_template_env(
                 python_version = python_version
             base_image = spec.image.format(spec_version=python_version)
     else:
-        base_image = docker_options.base_image
+        base_image = options.base_image
         logger.info(
             f"BentoML will not install Python to custom base images; ensure the base image '{base_image}' has Python installed."
         )
 
     return {
         **{
             f"__options__{k}": v
-            for k, v in bentoml_cattr.unstructure(docker_options).items()
+            for k, v in bentoml_cattr.unstructure(options).items()  # type: ignore
         },
-        **bentoml_cattr.unstructure(CustomizableEnv()),
-        **bentoml_cattr.unstructure(ReservedEnv(base_image, supported_architectures)),
+        **bentoml_cattr.unstructure(CustomizableEnv()),  # type: ignore
```

Comment:
probably nicer to write a `to_dict` method directly instead of using cattr. We only use cattr for attrs classes with nested attrs class fields